src/providers/definition.ts

```diff
@@ -5,13 +5,17 @@ import annotations from './annotations';
 import * as namespace from '../namespace';
 import * as outputWindow from '../results-output/results-doc';
 import * as replSession from '../nrepl/repl-session';
+import * as config from '../config';
 import { createConverter } from 'vscode-languageclient/lib/common/protocolConverter';
 import { getClient } from '../lsp/main';
 import { DefinitionRequest } from 'vscode-languageclient';
 
 const converter = createConverter(undefined, undefined);
 
-const definitionProviderOptions = { priority: ['lsp', 'repl'] };
+const definitionProviderOptions = {
+  replThenLsp: { priority: ['repl', 'lsp'] },
+  lspThenRepl: { priority: ['lsp', 'repl'] },
```

Comment:
@PEZ Do you think we could make this code + setting a bit more flexible by just letting them specify the priority array in the setting, rather than specifying a string that maps to a priority array? We could just explain that the providers will be tried in the order they specify in the setting.

Reply:
I suppose it's unlikely we'll add other providers, but if we did, it would be easier to just add a string to the enum of the setting, then add the definitionFunctions, and there wouldn't be a need to modify this mapping.

Reply:
Afaik there is no way to let the users specify order other than letting them specify the array. Then we would need to validate the entry and figure about error messages and such.

If there is a way to do this while ensuring the user can only specify our two options, I'd be most happy to change to that, for sure.

Reply:
Figured out how to specify the arrays in the enum. It renders a bit weird in the Settings view, but that's for the VS Code team to figure about, I think:

<img width="656" alt="image" src="https://user-images.githubusercontent.com/30010/175878304-75c2c527-80f0-4d56-8851-292a2747c544.png">


The user can still provide something bad, but the editor does a good job in highlighting this. (Also we had the same problem with the strings, anyway.)

<img width="537" alt="image" src="https://user-images.githubusercontent.com/30010/175878619-048e2056-b525-42fa-b7b6-2c3b11bbc92a.png">

Reply:
I agree with @bringe, having the array as configuration is more flexible. I'm hoping to add custom definition providers if they are useful, so it's good to keep the system as open as possible

Reply:
Yeah, I think we all agree. The only question was if we could provide a helpful settings UI. I'm all for open, but not if the Ux makes it hard to provide correct settings.

Reply:
As for custom definition providers... Maybe Calva isn't the place to add them? Please file a separate issue about that and we can discuss.

Reply:
I like the change! I was thinking more of an array of values from an enum, instead of an enum of arrays, but I don't know if that's possible with the settings. This is good too, though.

Reply:
Thanks for reviewing!